og.module

```diff
@@ -2630,6 +2630,21 @@ function og_role_permissions($roles = array()) {
 }
 
 /**
+ * Returns an array of all roles.
+ *
+ * @return array
+ *   An associative array with the role id as the key and the role name as
+ *   value.
+ */
+function og_get_all_roles() {
+  return db_select('og_role', 'ogr')
```

Comment:
this won't scale if we have multiple roles for each group (i.e. group that have their own roles)